spec/lib/phobos/producer_spec.rb

```diff
@@ -48,29 +48,71 @@ class TestProducer1
         expect(Phobos).to_not receive(:create_kafka_client)
       end
 
-      it 'publishes and delivers a list of messages' do
-        expect(kafka_client)
-          .to receive(:producer)
-          .with(TestProducer1.producer.regular_configs)
-          .and_return(producer)
+      context 'without cached producer' do
+        it 'publishes and delivers a list of messages' do
+          expect(kafka_client)
+            .to receive(:producer)
+            .with(TestProducer1.producer.regular_configs)
+            .and_return(producer)
 
-        expect(producer)
-          .to receive(:produce)
-          .with('message-1', topic: 'topic-1', key: 'key-1', partition_key: 'part-key-1')
+          expect(producer)
+            .to receive(:produce)
+            .with('message-1', topic: 'topic-1', key: 'key-1', partition_key: 'part-key-1')
 
-        expect(producer)
-          .to receive(:produce)
-          .with('message-2', topic: 'topic-2', key: 'key-2', partition_key: 'key-2')
+          expect(producer)
+            .to receive(:produce)
+            .with('message-2', topic: 'topic-2', key: 'key-2', partition_key: 'key-2')
 
-        expect(producer).to receive(:deliver_messages)
-        expect(producer).to receive(:shutdown)
-        expect(kafka_client).to_not receive(:close)
+          expect(producer).to receive(:deliver_messages)
+          expect(producer).to receive(:shutdown)
+          expect(kafka_client).to_not receive(:close)
 
-        subject.producer.publish_list([
-                                        { payload: 'message-1', topic: 'topic-1', key: 'key-1', partition_key: 'part-key-1' },
-                                        { payload: 'message-2', topic: 'topic-2', key: 'key-2' }
-                                      ])
+          subject.producer.publish_list([
+                                          { payload: 'message-1', topic: 'topic-1', key: 'key-1', partition_key: 'part-key-1' },
+                                          { payload: 'message-2', topic: 'topic-2', key: 'key-2' }
+                                        ])
+        end
+      end
+
+      context 'with cached producer' do
+        it 'publishes and delivers a list of messages twice' do
+          original_hash = Phobos.config.producer_hash
```

Comment:
similar to other comment I suggest a let

`let(:config) { Phobos.config.producer_hash.merge(persistent_connections: true) }`